x/twap/store.go

```diff
@@ -93,77 +91,68 @@ func (k Keeper) StoreHistoricalTWAP(ctx sdk.Context, twap types.TwapRecord) {
 func (k Keeper) pruneRecordsBeforeTimeButNewest(ctx sdk.Context, state types.PruningState) error {
 	store := ctx.KVStore(k.storeKey)
 
-	// Reverse iterator guarantees that we iterate through the newest per pool first.
-	// Due to how it is indexed, we will only iterate times starting from
-	// lastKeptTime exclusively down to the oldest record.
-	iter := store.ReverseIterator(
-		[]byte(types.HistoricalTWAPTimeIndexPrefix),
-		state.LastKeySeen)
-	defer iter.Close()
-
-	// We mark what (pool id, asset 0, asset 1) triplets we've seen.
-	// We prune all records for a triplet that we haven't already seen.
-	type uniqueTriplet struct {
-		poolId uint64
-		asset0 string
-		asset1 string
-	}
-	seenPoolAssetTriplets := map[uniqueTriplet]struct{}{}
-
 	var numPruned uint16
+	var lastPoolIdCompleted uint64
 
-	for ; iter.Valid(); iter.Next() {
-		timeIndexKey := iter.Key()
-		timeS, poolId, asset0, asset1, err := types.ParseFieldsFromHistoricalTimeKey(timeIndexKey)
+	for poolId := state.LastSeenPoolId; poolId > 0; poolId-- {
+		denoms, err := k.poolmanagerKeeper.RouteGetPoolDenoms(ctx, poolId)
 		if err != nil {
 			return err
 		}
 
-		poolKey := uniqueTriplet{
-			poolId,
-			asset0,
-			asset1,
+		// Notice, even if ranging over denomPairs takes us over the prune per block limit,
+		// we still continue to iterate over all denom pairs of the pool.
+		// This simplifies logic so that we can consider a pool "done" once we start it.
```

Comment:
Hrmm why is this important? Feels like we could strongly respect the limit just fine by revisiting the same pool next block, at no complexity change?

Reply:
Don't we need to store if we already skipped a record, as well as needing to skip to a certain index of denom pairs if we finish in between denom pairs?

Reply:
If you can think of a simpler way please lmk, but when I attempted this it turned into a bunch of additional logic that felt like overcomplication of something to meet a limit that is arbitrary anyway.

Reply:
The limits not arbitrary! Its to bound additional block time (e.g. so we don't accidentally get a new epoch time)

Reply:
The limit is not arbitrary, the value of the limit is.

Agreed on way forward in offline convo though!